### PR TITLE
Replacing docker.io occurrences with docker_proxy_cache

### PR DESCRIPTION
**Pull request by Akyiaa**
changes for tasks and pipelines 
Srtory: https://github.com/orgs/galasa-dev/projects/3/views/1?filterQuery=akyiaa&pane=issue&itemId=51275233

### CODE REVIEW

File: infrastructure/cicsk8s/galasa-prod/galasa-prod/server-testcatalog.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ spec:
                 - critical
       initContainers:
       - name: init-chown-data
-        image: docker.io/busybox:1.32.0
+        image: harbor.galasa.dev/docker_proxy_cache/busybox:1.32.0
```
Comment (jadecarino) [inline infrastructure/cicsk8s/galasa-prod/galasa-prod/server-testcatalog.yaml:46]: Can you add library in here please :)

**Comment by galasa-team** [comment]
Build successful

File: pipelines/tasks/helm.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ spec:
   steps:
   - name: helm
     workingDir: /workspace/git/$(params.context)
-    image: docker.io/alpine/helm:3.13.2
+    image: harbor.galasa.dev/docker_proxy_cache/alpine/helm:3.13.2
```
Comment (jadecarino) [inline pipelines/tasks/helm.yaml:24]: And here please

**Review by jadecarino — CHANGES_REQUESTED** [review]

**Comment by galasa-team** [comment]
Build successful

File: infrastructure/galasa-plan-b-lon02/argocd/argocd.yaml [hunk 3]
```diff
@@ -9909,7 +9909,7 @@ spec:
         - ""
         - --appendonly
         - "no"
-        image: redis:7.0.4-alpine
+        image: harbor.galasa.dev/docker_proxy_cache/library/redis:7.0.4-alpine
```
Comment (jadecarino) [inline infrastructure/galasa-plan-b-lon02/argocd/argocd.yaml:9912]: I think we can leave this one as it comes with the ArgoCD configuration - can you revert back please?

**Review by jadecarino — CHANGES_REQUESTED** [review]

**Review by jadecarino — APPROVED** [review]

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**